### PR TITLE
Fix gRPC encoding for BYTES payloads

### LINKED ISSUE #400: StringCodec not compliant with dataplane.proto -- gRPC failure
The StringCodec responds with a bytes string instead of an array of byte strings which causes serialization to fail for gRPC requests:

proto:
```
  // Representation for BYTES data type. The size must match what is
  // expected by the tensor's shape. The contents must be the flattened,
  // one-dimensional, row-major order of the tensor elements.
  repeated bytes bytes_contents = 8;

```

```
    def encode(cls, name: str, payload: List[str]) -> ResponseOutput:
        packed, shape = pack(map(_encode_str, payload))
        return ResponseOutput(
            name=name,
            datatype="BYTES",
            shape=shape,
            data=packed,
        )

```

Throws: 
```
ERROR:grpc._cython.cygrpc:Unexpected [TypeError] raised by servicer method [/inference.GRPCInferenceService/ModelInfer]
Traceback (most recent call last):
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi", line 680, in grpc._cython.cygrpc._handle_exceptions
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi", line 794, in _handle_rpc
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi", line 545, in _handle_unary_unary_rpc
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi", line 402, in _finish_handler_with_unary_response
  File "/Users/jacobneyer/.pyenv/versions/3.7.9/lib/python3.7/site-packages/mlserver/grpc/servicers.py", line 26, in _inner
    return await f(self, request, context)
  File "/Users/jacobneyer/.pyenv/versions/3.7.9/lib/python3.7/site-packages/mlserver/grpc/servicers.py", line 82, in ModelInfer
    response = ModelInferResponseConverter.from_types(result)
  File "/Users/jacobneyer/.pyenv/versions/3.7.9/lib/python3.7/site-packages/mlserver/grpc/converters.py", line 310, in from_types
    for output in type_object.outputs
  File "/Users/jacobneyer/.pyenv/versions/3.7.9/lib/python3.7/site-packages/mlserver/grpc/converters.py", line 310, in <listcomp>
    for output in type_object.outputs
  File "/Users/jacobneyer/.pyenv/versions/3.7.9/lib/python3.7/site-packages/mlserver/grpc/converters.py", line 345, in from_types
    type_object.data, datatype=type_object.datatype
  File "/Users/jacobneyer/.pyenv/versions/3.7.9/lib/python3.7/site-packages/mlserver/grpc/converters.py", line 291, in from_types
    return pb.InferTensorContents(**contents)
TypeError: 123 has type int, but expected one of: bytes
```

While 
```
    def encode(cls, name: str, payload: List[str]) -> ResponseOutput:
        packed, shape = pack(map(_encode_str, payload))
        return ResponseOutput(
            name=name,
            datatype="BYTES",
            shape=shape,
            data=[packed],
        )
```

serializes the response correctly

### PR DESCRIPTION
**Pull request by adriangonz**
Fixes #400

## Changelog
- Change current codec logic so that `BYTES` payloads don't get packed into a single bytes array, but instead is treated as a list of multiple bytes arrays.
- Add tests to ensure codecs work properly in both gRPC and REST

### CODE REVIEW

File: runtimes/alibi-explain/mlserver_alibi_explain/runtime.py [hunk 1]
```diff
@@ -58,7 +58,12 @@ async def explain_v1_output(self, request: InferenceRequest) -> Response:
         in the endpoint.
         """
         v2_response = await self.predict(request)
-        explanation = json.loads(v2_response.outputs[0].data.__root__)
+        output = v2_response.outputs[0]
+
+        # TODO: Does this default value make sense here?
+        explanation = {}
```
Comment (sakoush) [inline runtimes/alibi-explain/mlserver_alibi_explain/runtime.py:64]: perhaps we should raise an exception if we cannot recover the explanation dict?

File: runtimes/alibi-explain/mlserver_alibi_explain/common.py [hunk 2]
```diff
@@ -37,10 +37,15 @@ def convert_from_bytes(output: ResponseOutput, ty: Optional[Type]) -> Any:
     This utility function decodes the response from bytes string to python object dict.
     It is related to decoding StringCodec
     """
+    if output.shape != [1]:
+        # TODO: Does this default value make sense here?
+        return ""
```
Comment (sakoush) [inline runtimes/alibi-explain/mlserver_alibi_explain/common.py:42]: should we raise an exception?

**Review by sakoush — APPROVED** [review]
Lovely :) with regards to alibi runtime I feel that if the response is not according to spec that perhaps we could raise an exception instead of return empty dict that could confuse the consumer?

File: runtimes/alibi-explain/mlserver_alibi_explain/runtime.py (hunk 1 above)
Comment (adriangonz) [reply runtimes/alibi-explain/mlserver_alibi_explain/runtime.py:64]: Good point! Let me do that :+1:

**Pull request merged**